gg-pac-team3-back/src/models/UserCourseModel.mjs

```diff
@@ -0,0 +1,53 @@
+import admin from 'firebase-admin';
+
+class UserCourseModel {
+  constructor({
+    userId,
+    courseId,
+    enrolledAt = new Date(), // Default to current date if not provided
+    progress = 0, // Default to 0 if not provided
+    completed = false, // Default to false if not provided
+    earnedPoints = 0 // Default to 0 if not provided
```

Comment:
is there a difference between earnedPoints and progress?

Reply:
"earnedpoints" - mostly depend on quiz scores, while progress depends on the overall progress towards finishing the course.

Reply:
I would prefer instead of "progress", a field for "completedLessons" or something that tracks which quizzes/lessons they've completed and which are still incomplete. Calculating this value should be a task of the frontend based on what is marked as "completed", which the backend needs to keep track of.

Reply:
Okay, Eli. In the next PR, we will add the completedLessons and completedQuizzes fields, each storing objects that include the lessonId or quizId and their status (either 'completed' or 'incomplete') to track the progress of each lesson and quiz individually. We can introduce this feature in the next Pull Request. For now, we will leave this part as is because we need to research how to implement this method and will address it in the subsequent PR.